tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,264 @@
+    "mongodb::server::storage_engine: 'mmapv1'"
+]
+
+WIREDTIGER_MONGODB = [
+    "# Do not remove",
+    "mongodb::server::storage_engine: 'wiredTiger'"
+]
+
+
+@destructive
+class ScenarioPerformanceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer was able to change
+    the size or not
+
+    :expectedresults: The performance parameter should not be changed after upgrade.
+    """
+    @staticmethod
+    def _custom_hiera_file_creation(mongodb_type, tune_size="default"):
+        """
+        This function we will use to create a custom-hiera.yaml file
+        as per users requirement"
+        :param int mongodb_type: Use to select  MMPV1 or WiredTiger data.
+        :param str tune_size: tune size would be medium, large, extra-large,
+        extra-extra-large.
+        """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("\n".join(DEFAULT_CUSTOM_HIERA_DATA))
+            if mongodb_type:
+                fd.write("\n".join(MMPV1_MONGODB))
+            else:
+                fd.write("\n".join(WIREDTIGER_MONGODB))
+            if tune_size == "medium":
+                fd.write("\n".join(MEDIUM_TUNING_DATA))
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        self._custom_hiera_file_creation(mongodb_type, "medium")
+        try:
+            ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+            command_output = ssh.command('satellite-installer -s --disable-system-checks',
+                                         connection_timeout=1000).stdout
+            assert '  Success!' in command_output
+        except Exception:
+            test_case_status = "FAILED"
+            self._custom_hiera_file_creation(mongodb_type, "default")
+            ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+            command_output = ssh.command('satellite-installer -s --disable-system-checks',
+                                         connection_timeout=1000).stdout
+            assert '  Success!' in command_output
+            assert 'PASSED' == test_case_status
```

Comment:
This last assertion has no way of succeeding, as `test_case_status` has unconditionally assigned `"FAILED"`.

If your intention is to run some clean up code on failure, but still fail test, just re-raise caught exception:
```python
try:
    ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
    ...
except Exception:
    self._custom_hiera_file_creation(mongodb_type, "default")
    ... #  other cleanup code
    raise #  raise without arguments in exception handler re-raises this exception
```

Reply:
The purpose is only to restore the state if some kind of issue happens in the try part.
And the post-upgrade test depends on the pre-upgrade test if it gets complete successfully then and only the post-upgrade test gets to execute.